src/translator/processed_request_sentence.h

```
@@ -0,0 +1,280 @@
+
+    // End is simply begin + number of elements now
+    end_ = begin_ + size();
+  }
+
+  /// Storage size in bytes used by the range: storing size plus the contents.
+  size_t memSize() const {
+    return (                  //
+        sizeof(std::size_t)   // size_
+        + sizeof(T) * size()  // container elements
+    );
+  }
+
+  // Allows iterator based for loops.
+  const T *cbegin() const { return begin_; }
+  const T *cend() const { return end_; }
+
+  /// Number of elements in the range.
+  size_t size() const {
+    assert(sizeLoc_ != nullptr);
+    return *sizeLoc_;
+  }
+
+  // Allows index based for-loops using size()
+  const T &operator[](size_t i) const { return *(begin_ + i); }
+
+  /// Compute storageSize known only a vector. This is useful to compute the required-size to allocate before writing
+  /// onto a range.
+  static size_t storageSize(const std::vector<T> &v) {
+    return (                    //  Binary math follows
+        sizeof(std::size_t)     // The size variable written at the start.
+        + sizeof(T) * v.size()  // Vector elements.
+    );
+  }
+
+ private:
+  const size_t *sizeLoc_;  // &n
+
+  // Open interval [begin, end)
+  const T *begin_;  ///< &(v_{0})
+  const T *end_;    ///< &(v_{n})
+};
+
+/// Underlying flat sequential binary storage for entities in a ProcessedRequestSentence. Views are provided onto this
+/// storage to mimick ranges. This class allows for optimizing storage for L4-caches continuous binary representation
+/// requirements.
+class Storage {
+  friend class StorageIO;
+
+ public:
+  /// To be used in conjunction with delayedAllocate. A ProcessedRequestSentence is created uninitialized. Upon incoming
+  /// History, we compute the requireSize and allocate later with delayed allocate.
+  Storage() : data_{nullptr}, size_(0) {}
+
+  void delayedAllocate(size_t requiredSize) {
+    assert(!initialized());
+    size_ = requiredSize;
+    data_ = malloc(sizeof(char) * size_);
```

Any reason why C++ `New` is not used here?

---

This is untyped memory. Does [this answer](https://stackoverflow.com/a/184568/4565794) look like something I should be doing?

---

You can do that, I guess. I see no reason `data_` to be a `void *`, as it is used as a byte array (`char` or `unsigned char`) and then reinterpreted to the appropriate type. 

Also `malloc` can fail silently and you're not checking for that, whereas `new` will throw if it fails.